core/src/main/java/org/radargun/stages/test/TestStage.java

```diff
@@ -1,425 +1,164 @@
-         String propertyString = PropertyHelper.getPropertyString(properties.get(iterationProperty), this);
-         if (propertyString == null) {
-            throw new IllegalStateException("Unable to resolve iteration property '" + iterationProperty + "'.");
-         }
-         return propertyString;
-      }
-      return null;
-   }
-
-   protected Statistics createStatistics() {
-      return statisticsPrototype.copy();
-   }
-
-   protected boolean isFinished() {
-      return finished;
-   }
-
-   protected boolean isTerminated() {
-      return terminated;
-   }
-
-   public void setTerminated() {
-      this.terminated = true;
-   }
-
-   protected void setCompletion(Completion completion) {
-      this.completion = completion;
-   }
-
-   public Completion getCompletion() {
-      return completion;
-   }
-
-   public CountDownLatch getStartLatch() {
-      return startLatch;
-   }
-
-   public CountDownLatch getFinishLatch() {
-      return finishLatch;
-   }
-
-   public boolean useTransactions(String cacheName) {
-      return useTransactions.use(transactional, cacheName, transactionSize);
-   }
-
-   public abstract OperationLogic getLogic();
-
-   protected int getTestIteration() {
-      return testIteration;
-   }
-
-   protected static class StatisticsAck extends DistStageAck {
-      private final List<List<Statistics>> iterations;
-
-      protected StatisticsAck(SlaveState slaveState, List<List<Statistics>> iterations) {
-         super(slaveState);
-         this.iterations = iterations;
-      }
-   }
-
-   protected interface ResultRetriever<T> {
-      T getResult(Stressor stressor);
-      void mergeResult(T into, T that);
-   }
-
-   protected static class StatisticsResultRetriever implements ResultRetriever<Statistics> {
-      public StatisticsResultRetriever() {}
-
-      @Override
-      public Statistics getResult(Stressor stressor) {
-         return stressor.getStats();
-      }
-
-      @Override
-      public void mergeResult(Statistics into, Statistics that) {
-         into.merge(that);
-      }
-   }
-
-   private class TestTimeoutException extends RuntimeException {
-      public TestTimeoutException() {
-      }
-
-      public TestTimeoutException(Throwable cause) {
-         super(cause);
-      }
-   }
+        }
+        if (test != null) {
+            test.addResult(getTestIteration(), new Report.TestResult("Used threads", slaveResults, usedThreads.toString(), false));
+        } else {
+            log.info("No test name - results are not recorded");
+        }
+        if (failed) {
+            if (repeatCondition == null) {
+                return StageResult.FAIL;
+            } else {
+                return StageResult.BREAK;
+            }
+        }
+        if (checkRepeatCondition(aggregated)) {
+            return StageResult.SUCCESS;
+        } else {
+            return StageResult.BREAK;
+        }
+    }
+
+    protected void destroy() {
```

Comment:
Okay, I'll document the prepare and destroy methods :) So little clairvoyance in the crowd...